### DISCUSSION TITLE
I'm getting a "RedisClient::ReadTimeoutError" when trying to connect to AWS Redis

### DISCUSSION THREAD

**Discussion by daveomcd** [sidekiq/sidekiq#5820] [Q&A] [answered]
My apologies if this is the wrong location to post this.  Thought it made more sense than posting in "Issues."

I'm on ruby 3.2.0, sidekiq 7.0.6, and rails 7.0.2.

I am trying to connect my application to my redis cluster on AWS.  I have set my REDIS_URL with a password.  I'm currently getting this ReadTimeoutError.

```
bundle exec sidekiq -C config/sidekiq.yml -e production
2023-03-08T16:20:17.316Z pid=2388 tid=5rg INFO: Sidekiq 7.0.6 connecting to Redis with options {:size=>5, :pool_name=>"internal", :url=>"redis://:REDACTED@master.my-app-aws-redis-cluster.vxrtzy.use1.cache.amazonaws.com:6379/1"}
RedisClient::ReadTimeoutError
```

If I try the following though to test the information it seems to work.

```
redis-cli -h master.bane-aws-redis-cluster.vxrtzy.use1.cache.amazonaws.com -p 6379 -a '[REDACTED]' --tls ping
Warning: Using a password with '-a' or '-u' option on the command line interface may not be safe.
PONG
```
I have noticed I need the `--tls` flag there to get a PONG.  I've tried to figure out how to add that to my Rails application but so far haven't been able to get it to work.  Here's what is currently in my **initializers/sidekiq.rb** file.

```
schedule_file = "config/schedule.yml"

if File.exist?(schedule_file) && Sidekiq.server?
  Sidekiq::Cron::Job.load_from_hash YAML.load_file(schedule_file)
end

Sidekiq.configure_server do |config|
  config.redis = { ssl_params: { verify_mode: OpenSSL::SSL::VERIFY_NONE } }
end

Sidekiq.configure_client do |config|
  config.redis = { ssl_params: { verify_mode: OpenSSL::SSL::VERIFY_NONE } }
end
```

Is anyone able to help point me in a direction to trouble shoot this further or perhaps identify the issue?  Thanks for your time.

**Comment by mperham** [comment, marked as answer]
I think you need to use the “rediss” protocol to have your URL automatically activate TLS. 

On Wed, Mar 8, 2023, at 08:31, David McDonald wrote:
> 
> 
> My apologies if this is the wrong location to post this. Thought it made more sense than posting in "Issues."
> 
> I'm on ruby 3.2.0, sidekiq 7.0.6, and rails 7.0.2.
> 
> I am trying to connect my application to my redis cluster on AWS. I have set my REDIS_URL with a password. I'm currently getting this ReadTimeoutError.
> 
> `bundle exec sidekiq -C config/sidekiq.yml -e production
> 2023-03-08T16:20:17.316Z pid=2388 tid=5rg INFO: Sidekiq 7.0.6 connecting to Redis with options {:size=>5, :pool_name=>"internal", ***@***.***:6379/1"}
> RedisClient::ReadTimeoutError
`
> If I try the following though to test the information it seems to work.
> 
> `redis-cli -h master.bane-aws-redis-cluster.vxrtzy.use1.cache.amazonaws.com -p 6379 -a '[REDACTED]' --tls ping
> Warning: Using a password with '-a' or '-u' option on the command line interface may not be safe.
> PONG
`
> I have noticed I need the `--tls` flag there to get a PONG. I've tried to figure out how to add that to my Rails application but so far haven't been able to get it to work. Here's what is currently in my *initializers/sidekiq.rb* file.
> 
> `schedule_file = "config/schedule.yml"
> 
> if File.exist?(schedule_file) && Sidekiq.server?
>   Sidekiq::Cron::Job.load_from_hash YAML.load_file(schedule_file)
> end
> 
> Sidekiq.configure_server do |config|
>   config.redis = { ssl_params: { verify_mode: OpenSSL::SSL::VERIFY_NONE } }
> end
> 
> Sidekiq.configure_client do |config|
>   config.redis = { ssl_params: { verify_mode: OpenSSL::SSL::VERIFY_NONE } }
> end
`

**Reply by daveomcd** [reply]
Awesome! That worked. Thank so much for the help and all the great work!
